Application/EdFi.Ods.Api/Services/Metadata/Controllers/AggregateDependencyController.cs

```diff
@@ -1,146 +1,110 @@
+            try
             {
-                return Ok(GraphML.Create(_entityWithDataOperationGraphFactory.CreateGraph(includeTransformations: false)));
+                if (Request.Headers.Accept.Contains(new MediaTypeWithQualityHeaderValue(CustomMediaContentTypes.GraphML)))
+                {
+                    return Ok(CreateGraphML(_resourceLoadGraphFactory.CreateResourceLoadGraph()));
+                }
+
+                return Ok(GetGroupedLoadOrder(_resourceLoadGraphFactory.CreateResourceLoadGraph()));
             }
+            catch (NonAcyclicGraphException e)
+            {
+                // return a bad request if a circular reference occurs, with the path information of the reference.
+                string message = e.Message.Replace($"{Environment.NewLine}    is used by ", " -> ")
+                    .Replace(Environment.NewLine, " ");
 
-            var aggregateLoadOrders = SortEntities(_entityWithDataOperationGraphFactory.CreateGraph(includeTransformations: true))
-                                     .Where(
-                                          x => EntitySpecification.HasResourceRepresentation(x.Item1.Entity))
-                                     .ToList();
+                return BadRequest(message);
+            }
+        }
 
-            return Ok(aggregateLoadOrders.Select(x => AggregateLoadOrder.Create(x.Item1, x.Item2)));
+        // ReSharper disable once InconsistentNaming
+        private static GraphML CreateGraphML(BidirectionalGraph<Resource, AssociationViewEdge> resourceGraph)
+        {
+            return new GraphML
+            {
+                Id = "EdFi Dependencies",
+                Nodes = resourceGraph.Vertices
+                    .Select(r => new GraphMLNode {Id = GetNodeId(r)})
+                    .OrderBy(n => n.Id)
+                    .ToList(),
+                Edges = resourceGraph.Edges
+                    .Select(edge => new GraphMLEdge(GetNodeId(edge.Source), GetNodeId(edge.Target)))
+                    .GroupBy(x => x.Source)
+                    .OrderBy(g => g.Key)
+                    .SelectMany(g => g.OrderBy(x => x.Target))
+                    .ToList()
+            };
         }
 
-        private IEnumerable<Tuple<EntityWithDataOperation, int>> SortEntities(
-            BidirectionalGraph<EntityWithDataOperation, DataOperationEdge> executionGraph)
+        private static IEnumerable<ResourceLoadOrder> GetGroupedLoadOrder(
+            BidirectionalGraph<Resource, AssociationViewEdge> resourceGraph)
         {
-            var visited = new ConcurrentDictionary<EntityWithDataOperation, bool>(executionGraph.Vertices.ToDictionary(k => k, v => false));
-            var resolved = new List<Tuple<EntityWithDataOperation, int>>();
+            int groupNumber = 1;
 
-            int groupSet = 0;
+            var executionGraph = resourceGraph.Clone();
 
-            while (visited.Values.Any(x => x == false))
+            while (true)
```

Comment:
this always scares me even though you have the break down below, this could potentially cause an infinite loop. should you consider using a better conditional or change to a do while loop?